src/NotepadNext/docks/FolderAsWorkspaceDock.h

```diff
@@ -42,10 +44,22 @@ class FolderAsWorkspaceDock : public QDockWidget
 signals:
     void fileDoubleClicked(const QString &filePath);
 
+private slots:
+    void on_actionSaveHere_triggered();
+    void on_actionNewFolder_triggered();
+    void on_actionRename_triggered();
+    void on_actionDelete_triggered();
```

Comment:
Qt doesn't recommend the auto connect slots and I haven't used them elsewhere throughout the codebase, but for now it is fine. It can be cleaned up later.

Reply:
M.. why/where? `connectSlotsByName` has no such notices at any version.

Reply:
clazy is reporting it:

![image](https://github.com/dail8859/NotepadNext/assets/3694843/c1182f06-432d-41cc-8d4f-bd271db2d3a0)


https://github.com/KDE/clazy/blob/1.11/docs/checks/README-connect-by-name.md

Reply:
It is good scripting usage to automate creation of all these boilerplate as a lua script inside NotepadNext.
For some future version obviously
```lua
local luaxml = require 'LuaXML'

local file_name = nn.activeEditor.path:gsub('.ui$')
local cpp_editor = nn.findEditorByPath(file_name .. '.cpp')
local h_editor = nn.findEditorByPath(file_name .. '.h')

assert(cpp_editor and h_editor, 'Please open corresponding .cpp and .h')
local cpp_bookmark = cpp_editor:findNamedBookmark('ctor signals')
local h_bookmark = cpp_editor:findNamedBookmark('slots')
assert(cpp_bookmark and h_bookmark, 'Please set "ctor signals" and "slots" bookmarks')

local items = {}
local tree = xml.load(file_name .. '.ui')
tree:iterate(function(el)
  table.insert(items, el.tag('name'))
end, 'action', nil, nil, true)

local class_name = find_class_name(cpp_bookmark) -- TODO

for i,name in ipairs(items) do
  local method_name = 'on'.. name:gsub('^[a-z]', function(s) return s:upper() end) .. 'Triggered'
  local full_method = 'void '..method_name..'()'
  if not h_editor:findText(full_method) then
   h_bookmark:appendText(full_method .. ';\n');
   cpp_bookmark:append('connect(ui->'..name..', &QAction::triggered, this, &'.. class_name.. '::'..method_name..');')
   cpp_editor:append(full_method..'\n{\n}\n')
  end
end
```

Reply:
Interesting idea, though I don't see Lua as a long-term solution. It was originally just added for debugging/testing purposes

Reply:
I see lua as very good "Glue language" to customize and script stuff. I am going to use NotepadNext as very customizable light editor (instead of NotepadQQ and ZeroBraneStudio) on projects where I don't need big IDE like PyCharm